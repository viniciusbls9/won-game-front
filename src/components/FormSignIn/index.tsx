/* eslint-disable react/no-unescaped-entities */
import { Email, Lock } from 'styled-icons/material-outlined'
import Link from 'next/link'

import { Button } from 'components'
import TextField from 'components/TextField'
import * as S from './styles'

const FormSignIn = () => (
  <S.Wrapper>
    <form>
      <TextField
        name="email"
        placeholder="Email"
        type="email"
        icon={<Email />}
      />
      <TextField
        name="password"
        placeholder="Password"
        type="password"
        icon={<Lock />}
      />
      <S.ForgotPassword>Forgot your password?</S.ForgotPassword>

      <Button size="large" fullWidth>
        Sign in now
      </Button>

      <S.FormLink>
        Don't have an account?{' '}
        <Link href="/sign-up">
          <a>Sign up</a>
        </Link>
      </S.FormLink>
    </form>
  </S.Wrapper>
)

export default FormSignIn
