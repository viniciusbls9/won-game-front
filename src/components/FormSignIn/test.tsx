import '../../../.jest/match-media-mock'
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import FormSignIn from '.'

describe('<FormSignIn />', () => {
  it('should render the form', () => {
    renderWithTheme(<FormSignIn />)

    // verifique email field
    expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument()

    // verifique password field
    expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument()

    // verifique button
    expect(
      screen.getByRole('button', { name: /Sign in now/i })
    ).toBeInTheDocument()
  })

  it('should render the forgot password link', () => {
    renderWithTheme(<FormSignIn />)

    // verifique forgot password link
    expect(screen.getByText('Forgot your password?').closest('a'))
  })

  it('should render the text and link to sign up', () => {
    renderWithTheme(<FormSignIn />)

    // verifique text
    expect(screen.getByRole('link', { name: /sign up/i })).toBeInTheDocument()
    // verifique link
    expect(screen.getByText(/don't have an account\?/i)).toBeInTheDocument()
  })
})
