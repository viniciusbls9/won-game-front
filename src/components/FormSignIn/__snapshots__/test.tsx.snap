// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignIn /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    FormSignIn
  </h1>
</main>
`
